src/data/set/prod.lean

```diff
@@ -315,6 +315,9 @@ lemma mem_diagonal (x : α) : (x, x) ∈ diagonal α := by simp [diagonal]
 
 @[simp] lemma mem_diagonal_iff {x : α × α} : x ∈ diagonal α ↔ x.1 = x.2 := iff.rfl
 
+instance decidable_mem_diagonal [h : decidable_eq α] : decidable_pred (diagonal α) :=
```

Comment:
As seen in [set.decidable_mem_Ioi](https://leanprover-community.github.io/mathlib_docs/data/set/intervals/basic.html#set.decidable_mem_Ioi), it's would be more normal for such a lemma to use `decidable (x ∈ diagonal α)`.

Reply:
@linesthatinterlace It's common writing these as `decidable_pred` too, so long as you make sure to use a section of the operator (see my comment, which I wrote yesterday but forgot to submit the review, oops).

Reply:
Lean is able to figure out my particular use case for this either way, so I'm not really partial to one or the other.